### PR TITLE
DEV-4416 - Python client doesn't batch dataset upload and hits HTTP status 413 (Request entity too large)

### PR DESCRIPTION
**Pull request by jkgenser**
Added batching support and client test. Load testing this functionality is an integration test. So long as size of image x batch size is under the request size limit, batching will result in not hitting the size limit.

### CODE REVIEW

**Comment by jkgenser** [comment]
> I'm not sure we should make this image specific. I think we offer this for both image and pdf.

Updated to enable this.

File: indico/queries/datasets.py [hunk 1]
```diff
@@ -186,12 +186,14 @@ def __init__(
         wait: bool = True,
         from_local_images: bool = False,
         image_filename_col: str = "filename",
+        image_upload_batch_size: int = 20,
```
Comment (nkassis) [inline indico/queries/datasets.py:189]: I think this should just be batch size ignore image

Comment (jkgenser) [reply indico/queries/datasets.py]: Updated. Latest commit also has some additional type annotation that was missing on `UploadBatched`

Comment (jkgenser) [reply indico/queries/datasets.py]: nm... I get that you meant not just changing name. It clicked that there are other data types that can be uploaded. Added this and test for pdf.

File: indico/types/base.py [hunk 2]
```diff
@@ -17,26 +18,23 @@ def list_subtype(cls):
...
         return False
-    
+
     return (
-        (inspect.isclass(v) and issubclass(v, BaseType)) 
-        or v in [str, int, float, bool, JSONType] 
+        (inspect.isclass(v) and issubclass(v, BaseType))
+        or v in [str, int, float, bool, JSONType]
         or valid_type(list_subtype(v))
     )
```
Comment (jkgenser) [inline indico/types/base.py:27]: This is all formatting.

**Pull request merged**